Remotion/Data/DomainObjects/Persistence/Rdbms/RdbmsProvider.cs

```diff
@@ -39,14 +40,24 @@ public class RdbmsProvider : StorageProvider, IRdbmsProviderCommandExecutionCont
 
     public RdbmsProvider (
         RdbmsProviderDefinition definition,
+        string connectionString,
```

Comment:
IIRC I talked with @MichaelKetting about this and this was the solution we chose

Reply:
@a-ctor & @Frazerus Yeah, we probably mentioned and discarded the notion of using a readonly flag instead. if @Frazerus  and @a-ctor  decide to use the flag-option instead, that's okay, too. Neither solution feels perfect.

The reason I went with the argument check was because it's just a guardcondition and doesn't introduce branch logic per se either on the callsite or in the ctor. With the readonly flag, we have a boolean for this in the ctor (or further down when instantiating the connection) and thus more complex logic in the system as an entirety and make testing that much more complex.

The reason for doing the guard condition is mainly to force applications to not do something we did not expect with the system. It's possibly (though unlikely) that future refatorings would remove this constraint. Plus, there's a way to outsmart this right now, by overridng the sqlconnectionfactory and setting a connectionstring before rdbmsprovider gets to do its own thing.

So, yeah, my gut feeling prefers injecting the string instead of a flag because of its simplicity and straightforwardness and future concerns but an informed decision (see dispelling the concerns above) may overrule this.